snuba/utils/kafka/consumers/abstract.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Optional, Sequence, TypeVar
+
+from confluent_kafka import TopicPartition
+
+
+TMessage = TypeVar("TMessage")
+
+
+class Consumer(Generic[TMessage], ABC):
+    @abstractmethod
+    def subscribe(
+        self,
+        topics: Sequence[str],
+        on_assign: Optional[Callable[[Sequence[TopicPartition]], Any]] = None,
+        on_revoke: Optional[Callable[[Sequence[TopicPartition]], Any]] = None,
+    ) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def poll(self, timeout: Optional[float] = None) -> Optional[TMessage]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def commit(self, asynchronous: bool = True) -> Optional[Sequence[TopicPartition]]:
```

Comment:
This maps to the `confluent_kafka` return value, but I don't think it's good that the return value is different depending on the parameter values passed. I am thinking about changing it to a `Future` (so there is no distinction between the synchronous or asynchronous case) but that might take more change than is necessary/justifiable here.